### PR TITLE
baserow: unbreak the backend

### PR DESCRIPTION
**Pull request by RaitoBezarius**
###### Description of changes

Using @fabaff suggestions and @mweinelt patch, this unbreak the backend of Baserow.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by wegank** [comment]
As in #182692, you can perhaps add

```
sed 's/\[standard\]//' -i requirements/base.in requirements/base.txt
```

to postPatch.

**Comment by RaitoBezarius** [comment]
> 

That fixed it, indeed, but is it really a good idea?

File: pkgs/servers/baserow/default.nix [hunk 1]
```diff
@@ -122,8 +124,9 @@ buildPythonPackage rec {
     twisted
     unicodecsv
     uvicorn
+    watchgod
     zipp
-  ];
+  ] ++ uvicorn.optional-dependencies.standard;
```
Comment (mweinelt) [inline pkgs/servers/baserow/default.nix:129]: This only gets you the optional dependencies, but not uvloop itself. And `uvloop[standard]` requests both.

**Comment by mweinelt** [comment]
```
❯ ./result/bin/baserow
Traceback (most recent call last):
  File "/home/hexa/git/nixos/master/./result/bin/baserow", line 2, in <module>
    from baserow.manage import main
ModuleNotFoundError: No module named 'baserow'
```

**Comment by mweinelt** [comment]
Took a stab at getting this running, not sure about the django bits.

```diff
diff --git a/pkgs/servers/baserow/default.nix b/pkgs/servers/baserow/default.nix
index ceb3ff5d8da..1eb9b587a11 100644
--- a/pkgs/servers/baserow/default.nix
+++ b/pkgs/servers/baserow/default.nix
@@ -1,67 +1,21 @@
 { lib
-, buildPythonPackage
 , fetchFromGitLab
-, regex
-, service-identity
-, itsdangerous
-, requests
-, redis
-, channels
-, channels-redis
-, psycopg2
-, gunicorn
-, django-cors-headers
-, django-celery-email
-, advocate
-, django-storages
-, pillow
-, faker
-, uvicorn
-, twisted
-, django
-, drf-jwt
-, cryptography
-, tqdm
-, celery-redbeat
-, drf-spectacular
-, websockets
-, asgiref
-, antlr4-python3-runtime
-, psutil
-, dj-database-url
-, django-health-check
-, celery
-, unicodecsv
-, django-celery-beat
-, django-redis
-, zipp
-, boto3
-, cached-property
-, importlib-resources
-, zope_interface
-, freezegun
-, pyinstrument
-, responses
-, pytestCheckHook
-, setuptools
-, pytest-django
-, python
-, httpretty
-, pytest-unordered
-, watchgod
-, openapi-spec-validator }:
+, makeWrapper
+, python3
+}:
 
 let
 
-  baserow_premium = with python.pkgs; ( buildPythonPackage rec {
+  baserow_premium = with python3.pkgs; ( buildPythonPackage rec {
     pname = "baserow_premium";
     version = "1.10.2";
+    formt = "setuptools";
 
     src = fetchFromGitLab {
       owner = "bramw";
       repo = pname;
-      rev = version;
-      sha256 = "sha256-4BrhTwAxHboXz8sMZL0V68skgNw2D2/YJuiWVNe0p4w=";
+      rev = "refs/tags/${version}";
+      hash = "sha256-4BrhTwAxHboXz8sMZL0V68skgNw2D2/YJuiWVNe0p4w=";
     };
 
     sourceRoot = "source/premium/backend";
@@ -71,15 +25,16 @@ let
 
 in
 
-buildPythonPackage rec {
+with python3.pkgs; buildPythonPackage rec {
   pname = "baserow";
   version = "1.10.2";
+  format = "setuptools";
 
   src = fetchFromGitLab {
     owner = "bramw";
     repo = pname;
-    rev = version;
-    sha256 = "sha256-4BrhTwAxHboXz8sMZL0V68skgNw2D2/YJuiWVNe0p4w=";
+    rev = "refs/tags/${version}";
+    hash = "sha256-4BrhTwAxHboXz8sMZL0V68skgNw2D2/YJuiWVNe0p4w=";
   };
 
   sourceRoot = "source/backend";
@@ -91,6 +46,8 @@ buildPythonPackage rec {
     sed 's/\[standard\]//' -i requirements/base.in requirements/base.txt
   '';
 
+  nativeBuildInputs = [ makeWrapper ];
+
   propagatedBuildInputs = [
     advocate
     antlr4-python3-runtime
@@ -128,6 +85,12 @@ buildPythonPackage rec {
     zipp
   ] ++ uvicorn.optional-dependencies.standard;
 
+  postInstall = ''
+    wrapProgram $out/bin/baserow \
+      --prefix PYTHONPATH : "$PYTHONPATH" \
+      --prefix DJANGO_SETTINGS_MODULE : "baserow.api.settings"
+  '';
+
   checkInputs = [
     baserow_premium
     boto3
diff --git a/pkgs/top-level/all-packages.nix b/pkgs/top-level/all-packages.nix
index dc41958306e..9518e825e09 100644
--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -22478,7 +22478,7 @@ with pkgs;
 
   sabnzbd = callPackage ../servers/sabnzbd { };
 
-  baserow = python3Packages.callPackage ../servers/baserow { };
+  baserow = callPackage ../servers/baserow { };
 
   bftpd = callPackage ../servers/ftp/bftpd {};
 
```

**Comment by RaitoBezarius** [comment]
Will integrate your patch after lunch and test it.

**Comment by RaitoBezarius** [comment]
Huh, @onny how did you test this package? It seems like the frontend is not packaged here.

**Comment by RaitoBezarius** [comment]
Reading https://github.com/NixOS/nixpkgs/pull/177735 seems to indicate you were planning to PR module and frontend, we can push this PR, but I do not think it is a good idea to say that Baserow is packaged in NixOS when only the backend is in reality.
I will push my work and let you decide what you want to do with it, as you are the maintainer.

**Comment by onny** [comment]
Yes I have just packaged baserow backend as this was already quite some work for me as a NixOS beginner. I'm looking forward to package the frontend and a module as well. In the meantime it would be good to fix this package

**Comment by RaitoBezarius** [comment]
> Yes I have just packaged baserow backend as this was already quite some work for me as a NixOS beginner. I'm looking forward to package the frontend and a module as well. In the meantime it would be good to fix this package

In that case, I let you review this PR and merge it if it's fine, but it would be nice to have the frontend & module before the next NixOS release as it looks like to me a half-package IMHO, at least, it should be renamed to `baserow-api` or `baserow-server`.

**Comment by RaitoBezarius** [comment]
It would be nice to have NixOS tests to do simple curl tests.

**Comment by mweinelt** [comment]
This fixes the build which comes up in nixpkgs-review every so often, because this package uses a huge number of dependencies. I think we should merge this now and I hope @onny will find some time to do the broad strokes mentioned in https://github.com/NixOS/nixpkgs/pull/188872#issuecomment-1234797980.

**Pull request merged**